fhttp/http_utils.go

```
@@ -207,131 +208,127 @@ func removeTrailingPercent(s string) string {
 	return s
 }
 
-// generateStatus from string, format: status="503" for 100% 503s
-// status="503:20,404:10,403:0.5" for 20% 503s, 10% 404s, 0.5% 403s 69.5% 200s
-func generateStatus(status string) int {
-	lst := strings.Split(status, ",")
-	log.Debugf("Parsing status %s -> %v", status, lst)
+// parse input, i.e. "XX:20", using parseFunc
+func parseEntry(entry, about, input string, parseFunc func(string) (int64, error)) (int64, float64, error) {
+	l2 := strings.Split(entry, ":")
+	if len(l2) != 2 {
+		log.Warnf("Should have exactly 1 : in %s list %s -> %v", about, input, entry)
+		return -1, -1, errors.New("invalid format")
+	}
+	s, err := parseFunc(l2[0])
+	if err != nil {
+		log.Warnf("Bad input %s %v -> %v, not a number before colon", input, about, l2[0])
+		return -1, -1, errors.New("invalid format")
+	}
+	percStr := removeTrailingPercent(l2[1])
+	p, err := strconv.ParseFloat(percStr, 32)
+	if err != nil || p < 0 || p > 100 {
+		log.Warnf("Percentage is not a [0. - 100.] number in %v -> %v : %v %f", input, percStr, err, p)
+		return -1, -1, errors.New("invalid format")
+	}
+	return s, p, nil
+}
+
+func getWeightedValue(input string, weights []float32, values []int64) int64 {
+	res := 100. * rand.Float32()
+	for i, v := range weights {
+		if res <= v {
+			if i >= len(values) {
```

this test/if is new, why / when does this case happen ? (comment needed at least)
you could refactor to have one vector of pairs / struct instead of 2 arrays - or assert the 2 arrays have the same size instead of silently swallowing what must be a bug of usage)

---

added WeightValuePair struct instead of 2 arrays.